arch/risc-v/src/common/riscv_assert.c

```diff
@@ -57,6 +57,14 @@
 #  define CONFIG_BOARD_RESET_ON_ASSERT 0
 #endif
 
+/* Format output with register width and hex */
+
+#ifdef CONFIG_ARCH_RV32
+#  define PRIxREG "8"PRIx32
```

Comment:
let's use PRIxPTR to avoid chceck CONFIG_ARCH_RV32

Reply:
Yes, but we still need to specific the output width for rv32 (8) and rv64 (16) by it.

Reply:
yes, but we can check __LP64__ like what's is done in nuttx/arch/risc-v/include/types.h. The final goal is to remove CONFIG_ARCH_RV32 from Kconfig.

Reply:
OK

Reply:
OK.